modules/apps/portal-template/portal-template-freemarker/src/main/java/com/liferay/portal/template/freemarker/internal/FreeMarkerTemplateResourceLoader.java

```diff
@@ -48,7 +43,10 @@
 	@Modified
 	protected void activate(Map<String, Object> properties) {
 		init(
-			TemplateConstants.LANG_TYPE_FTL, _templateResourceParsers,
+			TemplateConstants.LANG_TYPE_FTL,
+			new HashSet<>(
```

Comment:
@W2jx This is not correct. The service tracker map always returns a copy of the list of tracked services, and you then copy list into a map. This means that any parse that are available and tracked by your registry won't be seen by these TemplateResourceLoader classes.

Please make sure you get the list from the registry dynamically. Also, no need to use a set because ServiceTrackerMap's value list will not duplicate.